extensions/wikia/PortableInfobox/tests/nodes/NodeComparisonTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+class NodeComparisonTest extends WikiaBaseTest {
+	protected function setUp() {
+		$this->setupFile = dirname( __FILE__ ) . '/../../PortableInfobox.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @covers       NodeComparison::getData
+	 * @covers       Node::getDataForChildren
+	 * @covers       Node::getChildNodes
+	 * @dataProvider dataProvider
+	 *
+	 * @param $markup
+	 * @param $params
+	 * @param $expected
+	 */
+	public function testGetData( $markup, $params, $expected ) {
+		$node = \Wikia\PortableInfobox\Parser\Nodes\NodeFactory::newFromXML( $markup, $params );
+
+		$this->assertEquals( $expected, $node->getData() );
+	}
+
+	public function dataProvider() {
+		return [
+			[ '<comparison><set><data source="1"/><data source="2"/></set></comparison>',
+			  [ '1' => 'one', '2' => 'two' ], [ 'value' => [
+				[ 'type' => 'set', 'data' => [ 'value' => [
+					[ 'type' => 'data', 'data' => [ 'label' => '', 'value' => 'one' ], 'isEmpty' => false,
+					  'source' => [ '1' ] ],
+					[ 'type' => 'data', 'data' => [ 'label' => '', 'value' => 'two' ], 'isEmpty' => false,
+					  'source' => [ '2' ] ]
+				] ], 'isEmpty' => false, 'source' => [ '1', '2' ] ] ]
+			  ] ],
+			[ '<comparison><set><data source="1"/><data source="2"/></set></comparison>',
+			  [ ], [ 'value' => [
+				[ 'type' => 'set', 'data' => [ 'value' => [
+					[ 'type' => 'data', 'data' => [ 'label' => '', 'value' => null ], 'isEmpty' => true,
+					  'source' => [ '1' ] ],
+					[ 'type' => 'data', 'data' => [ 'label' => '', 'value' => null ], 'isEmpty' => true,
+					  'source' => [ '2' ] ]
+				] ], 'isEmpty' => true, 'source' => [ '1', '2' ] ] ]
+			  ] ],
+			[ '<comparison><set><data source="1"/><data source="2"/></set></comparison>',
+			  [ '1' => 'one' ], [ 'value' => [
+				[ 'type' => 'set', 'data' => [ 'value' => [
+					[ 'type' => 'data', 'data' => [ 'label' => '', 'value' => 'one' ], 'isEmpty' => false,
+					  'source' => [ '1' ] ],
+					[ 'type' => 'data', 'data' => [ 'label' => '', 'value' => null ], 'isEmpty' => true,
+					  'source' => [ '2' ] ]
+				] ], 'isEmpty' => false, 'source' => [ '1', '2' ] ] ]
+			  ] ],
+		];
+	}
+
+	/**
+	 * @covers       NodeComparison::getRenderData
+	 * @covers       Node::getRenderDataForChildren
+	 * @dataProvider renderDataProvider
+	 *
+	 * @param $markup
+	 * @param $params
+	 * @param $expected
+	 */
+	public function testRenderData( $markup, $params, $expected ) {
+		$node = \Wikia\PortableInfobox\Parser\Nodes\NodeFactory::newFromXML( $markup, $params );
+
+		$this->assertEquals( $expected, $node->getRenderData() );
+	}
+
+	public function renderDataProvider() {
```

Comment:
Please add tests covering getRenderData in context of an empty node for all nodes